test/unit/dom/outerHTML.js

```diff
@@ -37,118 +45,52 @@ describe('dom: outerHTML', () => {
-          if (canPatchNativeAccessors) {
-            expect(elem.__outerHTML).to.equal('<slot><!--comment--></slot>');
-          }
-        } else if (type === 'root') {
-          expect(elem.outerHTML).to.equal('<_shadow_root_><!--comment--></_shadow_root_>');
-          if (canPatchNativeAccessors) {
-            expect(elem.__outerHTML).to.equal('<_shadow_root_><!--comment--></_shadow_root_>');
-          }
+          compare(elem, '<slot><!--comment--></slot>');
         }
       });
 
       it('should get a proper value from a node with two or more children', () => {
         elem.innerHTML = '<test></test><test2>text</test2><!--comment-->';
         if (type === 'host' || type === 'div') {
-          expect(elem.outerHTML).to.equal('<div><test></test><test2>text</test2><!--comment--></div>');
-          if (canPatchNativeAccessors && type === 'div') {
-            expect(elem.__outerHTML).to.equal('<div><test></test><test2>text</test2><!--comment--></div>');
-          }
+          compare(elem, '<div><test></test><test2>text</test2><!--comment--></div>');
         } else if (type === 'slot') {
-          expect(elem.outerHTML).to.equal('<slot><test></test><test2>text</test2><!--comment--></slot>');
-          if (canPatchNativeAccessors) {
-            expect(elem.__outerHTML).to.equal('<slot><test></test><test2>text</test2><!--comment--></slot>');
-          }
-        } else if (type === 'root') {
-          expect(elem.outerHTML).to.equal('<_shadow_root_><test></test><test2>text</test2><!--comment--></_shadow_root_>');
-          if (canPatchNativeAccessors) {
-            expect(elem.__outerHTML).to.equal('<_shadow_root_><test></test><test2>text</test2><!--comment--></_shadow_root_>');
-          }
+          compare(elem, '<slot><test></test><test2>text</test2><!--comment--></slot>');
         }
       });
 
       it('should set a proper value', () => {
         elem.innerHTML = '<div1><div2></div2><div3></div3></div1>';
         elem.childNodes[0].childNodes[0].outerHTML = '<div4></div4>';
         expect(elem.innerHTML).to.equal('<div1><div4></div4><div3></div3></div1>');
-        if (type !== 'host') {
-          if (canPatchNativeAccessors) {
-            expect(elem.__innerHTML).to.equal('<div1><div4></div4><div3></div3></div1>');
-          }
-        }
 
         elem.innerHTML = '<div1><div2></div2><div3></div3></div1>';
         elem.childNodes[0].outerHTML = '<div4></div4>';
         expect(elem.innerHTML).to.equal('<div4></div4>');
-
-        if (type !== 'host') {
-          if (canPatchNativeAccessors) {
-            expect(elem.__innerHTML).to.equal('<div4></div4>');
-          }
-        }
-
-        if (type === 'root') {
-          elem.innerHTML = '<div></div>';
-          // we can't change root with this, so nothing should happen
-          elem.outerHTML = '<p></p>';
-          expect(elem.outerHTML).to.equal('<_shadow_root_><div></div></_shadow_root_>');
```

Comment:
In its current state, yeah. It's far too confusing because we have 3 different environments:

1. Chrome native
2. WebKit
3. Standards compliant browsers that don't support v1 or v0